config/common.go

```diff
@@ -17,9 +17,12 @@ type commonOpts struct {
 	// Currently only relevant for HTTP/1.1 vs HTTP/2, since there's a potential
 	// change-of-protocol happening at our edge.
 	ForwardsProto string
+
+	// DEPRECATED: use TrafficPolicy instead.
+	Policy *policy
```

Comment:
Is this exposed in the config? Should we somehow pull it in `TrafficPolicy`?

Reply:
im not sure i understand the question.
is the concern that somebody might manually set Policy on the config struct vs using WithPolicy?
if that's a concern, we can always update the ApplyTCP, ApplyTLS, and ApplyHTTP functions to do the conversion instead. @nikolay-ngrok lmk if that would address the concern. I'm honestly leaning towards doing that anyway 🤷

Reply:
yeah, if ppl are passing in this field, they would expect it to work even if deprecated? or we should just return an error or something to make it clear.

Reply:
good call, my currently solution does not address all cases. I'll revert the `WithPolicy` function and change the implementations of `Apply<Proto>` to:
- convert `Policy` to `TrafficPolicy`
- print out the Deprecation warning 

This covers the cases where:
- the user uses `WithPolicy`
- the user creates the `Policy` struct without helpers and adds it to the config

Reply:
ok, change was made. Tests still run, policy still works on tunnel using old way of configuration